app/src/ui/history/commit-list-item.tsx

```diff
@@ -304,21 +326,155 @@ export class CommitListItem extends React.PureComponent<
+   * #desktop-app-contents, but something with electron/react makes it so
+   * anything appended outside the app is not accessible.
+   */
+  private trackCommitDrag(ghost: HTMLDivElement) {
+    const desktopAppContainer = document.getElementById('desktop-app-contents')
+    if (desktopAppContainer === null) {
+      log.warn('[onCommitMouseDown] - Could not locate desktop container!')
+      return
+    }
+
+    const { onDragStart, openBranchDropdown, selectedCommits } = this.props
+    let branchListItem: Element | null = null
+    let dragStarted = false
+    const copyMessageLabelElement = ghost.querySelector(
+      '.copy-message-label .branch-name'
+    )
+
+    // This is housed inside the trackCommitDrag method so we have its reference
+    // for removing the event listener. We could move it out but, then we would
+    // have move a lot of tracking out to the commit class and prefer to
+    // encapsulate all the non-react way of handling the drag event in one spot.
+    // If we make more things draggable, it may be prudent to refactor this into
+    // a draggable component.
+    function onMouseMove(moveEvent: MouseEvent) {
+      // Wait till user actually moves their mouse as opposed to clicking it.
+      if (!dragStarted && desktopAppContainer !== null) {
+        if (onDragStart !== undefined) {
+          onDragStart(selectedCommits)
+        }
+
+        desktopAppContainer.appendChild(ghost)
+        desktopAppContainer.classList.add('cherry-pick-mouse-over')
+        dragStarted = true
+
+        // Removes active status from commit selection
+        if (document.activeElement instanceof HTMLElement) {
+          document.activeElement.blur()
+        }
+      }
+
+      // place ghost next to mouse
+      const verticalOffset = __DARWIN__ ? 32 : 15
+      ghost.style.left = moveEvent.pageX + 0 + 'px'
+      ghost.style.top = moveEvent.pageY + verticalOffset + 'px'
+
+      // inspect element mouse is is hovering over
+      const elemBelow = document.elementFromPoint(
+        moveEvent.clientX,
+        moveEvent.clientY
+      )
+
+      // mouse left the screen
+      if (elemBelow === null) {
+        return
+      }
+
+      const branchDropdown = elemBelow.closest('.branch-button')
+      branchListItem = elemBelow.closest('.branches-list-item')
+
+      // We must be over the branch drop down button.
+      if (branchDropdown !== null) {
+        if (openBranchDropdown) {
+          openBranchDropdown()
+        }
+      }
+
+      // We must be over a branch.
+      if (branchListItem !== null) {
+        ghost.classList.add('over-branch')
+        // Grabbing branch name for copy label on windows implementation
+        const branchNameElement = branchListItem.querySelector('.name')
+        if (branchNameElement !== null && copyMessageLabelElement) {
+          copyMessageLabelElement.innerHTML = branchNameElement.innerHTML
+        }
+      } else if (branchListItem == null) {
```

Comment:
Probably a leftover, but not needed anymore, right? Or does the compiler complain?
```suggestion
      } else {
```

Reply:
Nope just left over from where I was comparing against null only before. Thx